File: examples/weather-api-python/_generators/endpoint/handler.py
---
to: {{ name | snake }}.py
desc: Create a new endpoint handler
---
from flask import jsonify


def get_{{ name | snake }}(city: str) -> dict:
    """Fetch {{ name | snake }} data for a city."""
    return {
        "city": city,
        "{{ name | snake }}": "placeholder",
    }


def {{ name | snake }}_handler(city: str):
    """Handle /{{ name | kebab }}/<city> requests."""
    data = get_{{ name | snake }}(city)
    return jsonify(data)

File: examples/weather-api-go/_generators/endpoint/handler.go
---
to: {{ name | snake }}.go
desc: Create a new endpoint handler
---
package main

import (
	"encoding/json"
	"net/http"
)

// {{ name | pascal }}Data holds {{ name | snake }} information.
type {{ name | pascal }}Data struct {
	City string `json:"city"`
	{{ name | pascal }} string `json:"{{ name | snake }}"`
}

func get{{ name | pascal }}(city string) {{ name | pascal }}Data {
	return {{ name | pascal }}Data{
		City: city,
		{{ name | pascal }}: "placeholder",
	}
}

func {{ name | camel }}Handler(w http.ResponseWriter, r *http.Request) {
	city := r.PathValue("city")
	if city == "" {
		http.Error(w, "city parameter required", http.StatusBadRequest)
		return
	}
	data := get{{ name | pascal }}(city)
	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(data)
}
